### ISSUE TITLE
DTS files are importing TS files - v3

### ISSUE THREAD

**Issue by Asha20** [LegendApp/legend-state#301]
Hi there, I'm migrating from 2.x to 3.0-alpha. I added a single line of code:

```ts
import type * as from from "@legendapp/state";
```

When I run `tsc` to type-check my code, I get a bunch of errors originating from `node_modules/@legendapp/state`. If I take a look at `node_modules/@legendapp/state/index.d.ts` as it is currently in `3.0.0-alpha.1`, here's what I see:

```ts
export { batch, beginBatch, endBatch } from './src/batching';
export { computed } from './src/computed';
export { linked } from './src/linked';
export { configureLegendState } from './src/config';
export { event } from './src/event';
// more imports...
```

The issue is that these files living in `src` are actually `.ts` files instead of `.d.ts` files. Because of this, when I run `tsc` those are pulled in for type-checking which causes type errors because for my own code I use a stricter `tsconfig.json` than the one `@legendapp/state` uses. Try setting `verbatimModuleSyntax` or `noUncheckedIndexedAccess` to `true` and you'll see what I mean.

Let me know if you need more info. Love the library by the way! So convenient and pleasant to work with.

**Comment by nouman2075** [comment]
#300

**Comment by FleetAdmiralJakob** [comment]
Forked the repo, and added these two typescript rules. This is what I have found:

![image](https://github.com/LegendApp/legend-state/assets/77451351/27fd0e6a-a753-4fb8-a58f-616d78403708)

**Comment by FleetAdmiralJakob** [comment]
@Asha20 Why are the files typechecked in the first place. I know that they are .ts files but I did exclude node_modules in my tsconfig
![image](https://github.com/LegendApp/legend-state/assets/77451351/cc15f5c1-046d-4584-81cd-7b46e966c190)

**Comment by Asha20** [comment]
Looks like that's just how TypeScript works: https://github.com/microsoft/TypeScript/issues/41883#issuecomment-1758692340

I also intuitively expected being able to use `exclude` to side-step the issue, but apparently it doesn't work.

**Comment by FleetAdmiralJakob** [comment]
I have a question: What if before the code is published on npm, it gets fully compiled down to js so typescript does not check it? Is that a solution?

**Comment by jmeistrich** [comment]
Sorry for the delay on this! I was mostly offline for a few days while going to an out of town family thing. I just released a new alpha.2 which should fix this.

I had at some point in the `@next` versions changed the build to include the src folder alongside the built files, and that worked great in every environment I tested, and nobody testing that version seemed to have any issues with it. But it seems like some environments do check against node_modules. I don’t know why it would have all those failures though… I’ve seen other packages include their src folder with no issues. Package building is hard 😅

So I removed `src` from the published package and changed the build to use `tsup` instead of rollup, which reduced file sizes too. So I think that should fix all of this. Let me know if it’s working better for you in alpha.2?
